Jenkinsfile

```diff
@@ -1,65 +1,135 @@
+        GITHUB_ORG = 'aml-org'
+        GITHUB_REPO = 'scala-common'
+    }
+    stages {
+        stage('Test') {
+            steps {
+                script {
+                    lastStage = env.STAGE_NAME
+                    sh 'sbt clean coverage test coverageAggregate'
+                }
             }
-          }
         }
-      }
-    }
-    stage('Publish') {
-      when {
-        branch 'master'
-      }
-      steps {
-        wrap([$class: 'AnsiColorBuildWrapper', 'colorMapName': 'XTerm']) {
-          sh '''
-              echo "about to publish in sbt"
-              sbt commonJS/publish
-              sbt commonJVM/publish
-              echo "sbt publishing successful"
-          '''
+        stage('Coverage') {
+            when {
+                anyOf {
+                    branch 'master'
+                }
+            }
+            steps {
+                withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: 'sonarqube-official', passwordVariable: 'SONAR_SERVER_TOKEN', usernameVariable: 'SONAR_SERVER_URL']]) {
+                    script {
+                        lastStage = env.STAGE_NAME
+                        sh 'sbt -Dsonar.host.url=${SONAR_SERVER_URL} -Dsonar.login=${SONAR_SERVER_TOKEN} sonarScan'
+                    }
+                }
+            }
+        }
+        stage('Publish') {
+            when {
+                branch 'master'
+            }
+            steps {
+                wrap([$class: 'AnsiColorBuildWrapper', 'colorMapName': 'XTerm']) {
+                    script {
+                        lastStage = env.STAGE_NAME
+                        sh '''
+                               echo "about to publish in sbt"
+                               sbt commonJS/publish
+                               sbt commonJVM/publish
+                               echo "sbt publishing successful"
+                           '''
+                    }
+                }
+            }
+        }
+        stage('Tag version') {
+            when {
+                branch 'master'
+            }
+            steps {
+                withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: 'github-salt', passwordVariable: 'GITHUB_PASS', usernameVariable: 'GITHUB_USER']]) {
+                    script {
+                        lastStage = env.STAGE_NAME
+                        def version = sbtArtifactVersion("commonJVM")
+                        tagCommitToGithub(version)
+                    }
+                }
+            }
         }
-      }
     }
-    stage('Tag version') {
-      when {
-        branch 'master'
-      }
-      steps {
-        withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: 'github-salt', passwordVariable: 'GITHUB_PASS', usernameVariable: 'GITHUB_USER']]) {
-          script {
-            def version = sbtArtifactVersion("commonJVM")
-            tagCommitToGithub(version)
-          }
+    post {
```

Comment:
This is a cleaner approach https://github.com/mulesoft/apb/blob/develop/Jenkinsfile#L137